db/schema.go

```diff
@@ -154,35 +149,51 @@ func createTriggerFunc(tx *pgx.Tx) error {
 	return err
 }
 
-func getTablesToRegister(conn *pgx.Conn, schemas []string, excludeTables []string) ([]string, error) {
-	exclude := make(map[string]struct{})
-	// TODO: support patterns
-	for _, t := range excludeTables {
-		exclude[t] = struct{}{}
-	}
+// GenerateTablesList using the includes and excludes list. If no tables are specified in the includes list,
+// obtain the complete list from Postgres using the supplied schemas. If any of the included tables are listed
+// as excluded, remove them from the list.
+func GenerateTablesList(conn *pgx.Conn, schemas, includeTables, excludeTables []string) ([]string, error) {
+	tableRegister := make(map[string]bool)
 
-	var tables []string
-	for _, schema := range schemas {
-		rows, err := conn.Query(`
-		SELECT tablename FROM pg_catalog.pg_tables WHERE schemaname = $1`, schema,
-		)
-		if err != nil {
-			return nil, err
+	if len(includeTables) > 0 {
+		for _, table := range includeTables {
+			tableRegister[table] = true
 		}
-
-		for rows.Next() {
-			var t string
-			err = rows.Scan(&t)
+	} else {
+		for _, schema := range schemas {
+			rows, err := conn.Query(`
+		SELECT tablename FROM pg_catalog.pg_tables WHERE schemaname = $1`, schema,
+			)
 			if err != nil {
-				return nil, nil
+				return nil, err
 			}
 
-			if _, ok := exclude[t]; !ok {
-				tables = append(tables, fmt.Sprintf("%s.%s", schema, t))
+			for rows.Next() {
+				var t string
+				err = rows.Scan(&t)
+				if err != nil {
+					return nil, nil
```

Comment:
this needs to be `return nil, err`

Reply:
ah nice! thanks. that was a copy and paste 😂

Reply:
done